e2e-tests/portal/cypress/e2e/journeys/admin/restrict-access/restrict-access.spec.js

```diff
@@ -1,40 +1,53 @@
 const relative = require('../../../relativeURL');
-const MOCK_USERS = require('../../../../fixtures/users');
-
-const validUsers = ['ADMIN', 'UKEF_OPERATIONS'];
-const invalidUsers = ['BANK1_MAKER1', 'BANK1_CHECKER1', 'EDITOR'];
+const {
+  ADMIN,
+  UKEF_OPERATIONS,
+  BANK1_MAKER1,
+  BANK1_CHECKER1,
+  EDITOR,
+  BANK1_READ_ONLY,
+} = require('../../../../fixtures/users');
 
 context('Only allow authorised users to access admin pages', () => {
-  it('should allow Admins access to restricted pages', () => {
-    const user = MOCK_USERS[validUsers[0]];
-    cy.login(user);
-    cy.visit('/admin/users/');
-    cy.url().should('eq', relative('/admin/users/'));
-  });
+  const authorisedRoles = [{
+    roleName: 'Admins',
+    userWithRole: ADMIN,
+  }, {
+    roleName: 'UKEF Operations',
+    userWithRole: UKEF_OPERATIONS,
+  }];
 
-  it('should allow UKEF Operations access to restricted pages', () => {
-    const user = MOCK_USERS[validUsers[1]];
-    cy.login(user);
-    cy.visit('/admin/users/');
-    cy.url().should('eq', relative('/admin/users/'));
+  authorisedRoles.forEach(({ roleName, userWithRole }) => {
+    it(`should allow ${roleName} access to restricted pages`, () => {
+      cy.login(userWithRole);
+      cy.visit('/admin/users/');
+      cy.url().should('eq', relative('/admin/users/'));
+    });
   });
 
-  it('should NOT allow Makers access to restricted pages', () => {
-    const user = MOCK_USERS[invalidUsers[0]];
-    cy.login(user);
-    cy.visit('/admin/users/');
-    cy.url().should('eq', relative('/dashboard/deals/0'));
-  });
-  it('should NOT allow Checkers access to restricted pages', () => {
-    const user = MOCK_USERS[invalidUsers[1]];
-    cy.login(user);
-    cy.visit('/admin/users/');
-    cy.url().should('eq', relative('/dashboard/deals/0'));
-  });
-  it('should NOT allow Editors access to restricted pages', () => {
-    const user = MOCK_USERS[invalidUsers[2]];
-    cy.login(user);
-    cy.visit('/admin/users/');
-    cy.url().should('eq', relative('/login'));
+  const unauthorisedRoles = [{
+    roleName: 'Makers',
+    userWithRole: BANK1_MAKER1,
+    expectedRedirectLocation: '/dashboard/deals/0',
+  }, {
+    roleName: 'Checkers',
+    userWithRole: BANK1_CHECKER1,
+    expectedRedirectLocation: '/dashboard/deals/0',
+  }, {
+    roleName: 'Editors',
```

Comment:
No such role as a editor, unless the scope here is to test with a fictitious role?

Reply:
This test with this role already existed, I only refactored it
Our mock admin user has the "editor" role so it looks like it existed at some point in DTFS's history
We've got a separate sub-task to remove any reference to it - we should check the dev / staging / live databases as well to make sure no-one has it as a role